msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/errors/Error.kt

```diff
@@ -0,0 +1,132 @@
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.nativeauth.statemachine.errors
+
+import com.microsoft.identity.nativeauth.statemachine.results.ResetPasswordResendCodeResult
+import com.microsoft.identity.nativeauth.statemachine.results.ResetPasswordSubmitCodeResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignInResendCodeResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignInSubmitCodeResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignUpResendCodeResult
+import com.microsoft.identity.nativeauth.statemachine.results.SignUpSubmitCodeResult
+
+/**
+ * ErrorTypes class holds the possible error type values that are shared between the errors
+ * returned from each flow.
+ */
+internal class ErrorTypes () {
+    companion object {
+        /*
+         * The BROWSER_REQUIRED value indicates that the server requires more/different authentication mechanisms than the client is configured to provide.
+         * The flow should be restarted with a browser, by calling [com.microsoft.identity.client.IPublicClientApplication.acquireToken]
+         */
+        const val BROWSER_REQUIRED = "browser_required"
+
+        /*
+         * The INVALID_CODE value indicates the verification code provided by user is incorrect.
+         * The code should be re-submitted.
+         */
+        const val INVALID_CODE = "invalid_code"
+
+        /*
+         * The USER_NOT_FOUND value indicates there was no account found with the provided email.
+         * The flow should be restarted.
+         */
+        const val USER_NOT_FOUND = "user_not_found"
+
+        /*
+         * The INVALID_PASSWORD value indicates the new password provided by the user was not accepted by the server.
+         * The password should be re-submitted.
+         */
+        const val INVALID_PASSWORD = "invalid_password"
+    }
+}
+
+/**
+ * Error is a base class for all errors present in the Native Auth.
+ * @param errorType the error type value of the error that occurred
+ * @param error the error returned by the authentication server.
+ * @param errorMessage the error message returned by the authentication server.
+ * @param correlationId a unique identifier for the request that can help in diagnostics.
+ * @param errorCodes a list of specific error codes returned by the authentication server.
+ * @param exception an internal unexpected exception that happened.
+ */
+sealed class Error(
```

Comment:
@GBakolas should this not be a normal class, rather than a sealed?

Reply:
Yes it should be